src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -305,8 +295,13 @@ public class XmlConvert
         }
 
         private const int EncodedCharLength = 7; // ("_xFFFF_".Length);
-        private static volatile Regex? s_encodeCharPattern;
-        private static volatile Regex? s_decodeCharPattern;
+
+        [RegexGenerator("_[Xx][0-9a-fA-F]{4}(?:_|[0-9a-fA-F]{4}_)")]
```

Comment:
You changed this one as well, from `"_[Xx]([0-9a-fA-F]{4}|[0-9a-fA-F]{8})_")` which looks fine